parcels/nodes/PyNode.py

```diff
@@ -0,0 +1,601 @@
+        :returns boolean, if :arg other is ordered before-or-at the position of this
+        """
+        if type(self) is not type(other):
+            err_msg = "This object and the other object (type={}) do note have the same type.".format(str(type(other)))
+            raise AttributeError(err_msg)
+        return self.data.id <= other.data.id
+
+    def __gt__(self, other):
+        """
+        :arg other: another Node object
+        :returns boolean, if :arg other is ordered after the position of this
+        """
+        if type(self) is not type(other):
+            err_msg = "This object and the other object (type={}) do note have the same type.".format(str(type(other)))
+            raise AttributeError(err_msg)
+        return self.data.id > other.data.id
+
+    def __ge__(self, other):
+        """
+        :arg other: another Node object
+        :returns boolean, if :arg other is ordered after-or-at the position of this
+        """
+        if type(self) is not type(other):
+            err_msg = "This object and the other object (type={}) do note have the same type.".format(str(type(other)))
+            raise AttributeError(err_msg)
+        return self.data.id >= other.data.id
+
+    def __repr__(self):
+        """
+        :returns a byte-like representation of a Node
+        """
+        return '<%s.%s object at %s>' % (
+            self.__class__.__module__,
+            self.__class__.__name__,
+            hex(id(self))
+        )
+
+    def __str__(self):
+        """
+        returns a text-like representation of a Node
+        """
+        return "Node(p: {}, n: {}, id: {}, d: {})".format(repr(self.prev), repr(self.next), self.data.id, repr(self.data))
+
+    def __sizeof__(self):
+        """
+        :returns the byte size of this object, INCLUDING the size of its containing object
+        """
+        obj_size = sys.getsizeof(object)+sys.getsizeof(object)
+        if self.data is not None:
+            obj_size += sys.getsizeof(self.data)
+        return obj_size
+
+    def set_prev(self, prev):
+        """
+        (mandatory) setter-function for the previous Node object
+        """
+        self.prev = prev
+
+    def set_next(self, next):
+        """
+        (mandatory) setter-function for the next Node object
+        """
+        self.next = next
+
+    def set_data(self, data):
+        """
+        (mandatory) setter-function for the data payload of this Node
+        """
+        self.data = data
+
+    def reset_data(self):
+        """
+        this function resets (i.e. nullifies / nonifies) the the Node's data payload
+        """
+        self.data = None
+
+    def reset_prev(self):
+        """
+        this function resets (i.e. nullifies / nonifies) the the Node's predecessor
+        """
+        self.prev = None
+
+    def reset_next(self):
+        """
+        this function resets (i.e. nullifies / nonifies) the the Node's successor
+        """
+        self.next = None
```

Comment:
Not sure that `reset` is the best phrase here in these three functions. Suggests to me that it's overwritten with another value. Better to use `remove`?